src/stdatamodels/jwst/datamodels/tests/test_models.py

```diff
@@ -373,3 +374,36 @@ def test_ramp_model_zero_frame_by_dimensions():
 
     with datamodels.RampModel(dims) as ramp:
         assert ramp.zeroframe.shape == zdims
+
+
+def test_dq_def_roundtrip(tmp_path):
+    """
+    Open a MaskModel with a defined DQ array and dq_def that modifies the
+    DQ array on open.  Save the model to a new name.  Open the new model
+    to verify the DQ array is equal to the original DQ array.  This validates
+    that the dynamic_mask is invertible and computes the same values.
+    """
+    bname = "jwst_nircam_mask_ref.fits"
+    nbname = bname.replace("ref", "ref_dummy")
+
+    fname = os.path.join(ROOT_DIR, bname)
+    new_fname = os.path.join(tmp_path, nbname)
+
+    diff = None
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        # Open MaskModel, making sure it has non None dq and dq_def, then save it
+        with MaskModel(fname) as mask:
+            mask.save(new_fname)
```

Comment:
Is this supposed to fail if it as a `None` `dq` or `dq_def`? or is this just supposed to prevent the warning from appearing?

Reply:
This is just to prevent the warnings from appearing.